test/functional/tests.py

```diff
@@ -648,5 +740,44 @@ def test_bitswap_unwant(self):
         result = self.api.bitswap_unwant(key='QmZTR5bcpQD7cFgTorqxZDYaew1Wqgfbd2ud9QqGPAkK2V')
         self.assertTrue(result is not None)
 
+# Run test for `.shutdown()` only as the last test in CI environments – it would be to annoying
+# during normal testing
+@skipIfOffline()
+@skipUnlessCI()
+@pytest.mark.last
+@pytest.mark.xfail(reason="Daemon does not shut down on Travis for some reason")
+class IpfsApiShutdownTest(unittest.TestCase):
+    def setUp(self):
+        self.api = ipfsapi.Client()
+    
+    def _is_ipfs_running(self):
+        # Test only running on Linux CI – use procfs directly
+        for pid in filter(lambda n: n.isdigit(), os.listdir("/proc")):
```

Comment:
Actually it works on Linux and on Linux only. When I initally wrote it that was OK, because I only intended for it to be run on Travis (where it still doesn't work for unknown reasons…).
Then I added the `./test.sh` script for testing this test and later decided to add it to a commit – which means the test isn't Travis-exclusive anymore.

I think I'll rewrite this to be cross-platform (including Windows). Not doing so will just result in technical depth that needs to be cleaned up down the road. 😉